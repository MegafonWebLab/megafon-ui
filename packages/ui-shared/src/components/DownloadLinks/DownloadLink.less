@import '~@megafon/ui-core/styles/base.less';
@root: e('.mfui-beta-download-link');

@{root} {
    display: flex;

    &__icon {
        flex-shrink: 0;
        width: 40px;
        height: 40px;
        margin-right: 22px;

        @media @mobile {
            width: 32px;
            height: 32px;
        }
    }

    &__icon-svg {
        width: inherit;
        height: inherit;
    }

    &__info {
        margin: 0;
        .small();
        color: rgba(@freshAsphalt, 0.5);
        text-transform: uppercase;
    }
}
