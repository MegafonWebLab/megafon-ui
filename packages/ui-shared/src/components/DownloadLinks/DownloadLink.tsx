import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate, TextLink } from '@megafon/ui-core';
import './DownloadLink.less';
import DownloadIcon from 'icons/Basic/32/Download_32.svg';

export interface IDownloadLink {
    /** Адресы ссылки */
    href: string;
    /** Текст ссылки */
    text: string;
    /** Расширение файла */
    extension: string;
    /** Размер файла */
    fileSize: string;
    /** Дополнительный класс для компонента */
    className?: string;
    /** Обработчик клика по ссылке */
    onClick?: (e: React.SyntheticEvent<EventTarget>) => void;
}

const cn = cnCreate('mfui-beta-download-link');
const DownloadLink: React.FC<IDownloadLink> = ({ href, text, extension, fileSize, onClick, className }) =>
   (
        <div className={cn([className])}>
            <div className={cn('icon')}>
                <DownloadIcon />
            </div>
            <div>
                <TextLink className={cn('link')} href={href} onClick={onClick} download>{text}</TextLink>
                <p className={cn('info')}>
                    {`${extension}, ${fileSize}`}
                </p>
            </div>
        </div>
   );

DownloadLink.propTypes = {
    href: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    extension: PropTypes.string.isRequired,
    fileSize: PropTypes.string.isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func,
};

export default DownloadLink;
