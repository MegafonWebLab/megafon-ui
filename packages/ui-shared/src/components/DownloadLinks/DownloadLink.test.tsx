import * as React from 'react';
import { shallow } from 'enzyme';
import DownloadLink, { IDownloadLink } from './DownloadLink';

const props: IDownloadLink = {
    href: 'href',
    text: 'text',
    extension: 'PDF',
    fileSize: '500 MB',
};

describe('DownloadLink', () => {
    it('render component', () => {
        const wrapper = shallow(<DownloadLink {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render with custom classname', () => {
        const wrapper = shallow(<DownloadLink {...props} className="custom-classname" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onClick props', () => {
        const onClick = jest.fn();
        const wrapper = shallow(<DownloadLink {...props} onClick={onClick} />);
        const link = wrapper.find('TextLink');

        link.simulate('click');
        expect(onClick).toBeCalled();
    });
});
