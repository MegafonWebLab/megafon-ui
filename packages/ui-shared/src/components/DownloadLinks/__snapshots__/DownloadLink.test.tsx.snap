// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadLink render component 1`] = `
<div
  className="mfui-beta-download-link"
>
  <div
    className="mfui-beta-download-link__icon"
  >
    <test-file-stub
      className="mfui-beta-download-link__icon-svg"
    />
  </div>
  <div>
    <TextLink
      className="mfui-beta-download-link__link"
      color="blue"
      download={true}
      href="href"
      underlineStyle="solid"
      underlineVisibility="hover"
    >
      text
    </TextLink>
    <p
      className="mfui-beta-download-link__info"
    >
      PDF, 500 MB
    </p>
  </div>
</div>
`;

exports[`DownloadLink render with custom classname 1`] = `
<div
  className="mfui-beta-download-link custom-classname"
>
  <div
    className="mfui-beta-download-link__icon"
  >
    <test-file-stub
      className="mfui-beta-download-link__icon-svg"
    />
  </div>
  <div>
    <TextLink
      className="mfui-beta-download-link__link"
      color="blue"
      download={true}
      href="href"
      underlineStyle="solid"
      underlineVisibility="hover"
    >
      text
    </TextLink>
    <p
      className="mfui-beta-download-link__info"
    >
      PDF, 500 MB
    </p>
  </div>
</div>
`;
